proto/ibc/core/connection/v1/connection.proto

```diff
@@ -17,37 +17,41 @@ import "ibc/core/commitment/v1/commitment.proto";
 message ConnectionEnd {
   option (gogoproto.goproto_getters) = false;
   // client associated with this connection.
-  string client_id = 1 [ (gogoproto.moretags) = "yaml:\"client_id\"" ];
+  string client_id = 1 [(gogoproto.moretags) = "yaml:\"client_id\""];
   // IBC version which can be utilised to determine encodings or protocols for
   // channels or packets utilising this connection.
   repeated Version versions = 2;
   // current state of the connection end.
   State state = 3;
   // counterparty chain associated with this connection.
-  Counterparty counterparty = 4 [ (gogoproto.nullable) = false ];
+  Counterparty counterparty = 4 [(gogoproto.nullable) = false];
   // delay period that must pass before a consensus state can be used for
   // packet-verification NOTE: delay period logic is only implemented by some
   // clients.
-  uint64 delay_period = 5 [ (gogoproto.moretags) = "yaml:\"delay_period\"" ];
+  uint64 delay_time_period = 5 [(gogoproto.moretags) = "yaml:\"delay_period\""];
+  // delay height period associated with this connection.
+  uint64 delay_height_period = 6 [(gogoproto.moretags) = "yaml:\"delay_height_period\""];
 }
 
 // IdentifiedConnection defines a connection with additional connection
 // identifier field.
 message IdentifiedConnection {
   option (gogoproto.goproto_getters) = false;
   // connection identifier.
-  string id = 1 [ (gogoproto.moretags) = "yaml:\"id\"" ];
+  string id = 1 [(gogoproto.moretags) = "yaml:\"id\""];
   // client associated with this connection.
-  string client_id = 2 [ (gogoproto.moretags) = "yaml:\"client_id\"" ];
+  string client_id = 2 [(gogoproto.moretags) = "yaml:\"client_id\""];
   // IBC version which can be utilised to determine encodings or protocols for
   // channels or packets utilising this connection
   repeated Version versions = 3;
   // current state of the connection end.
   State state = 4;
   // counterparty chain associated with this connection.
-  Counterparty counterparty = 5 [ (gogoproto.nullable) = false ];
-  // delay period associated with this connection.
-  uint64 delay_period = 6 [ (gogoproto.moretags) = "yaml:\"delay_period\"" ];
+  Counterparty counterparty = 5 [(gogoproto.nullable) = false];
+  // delay time period associated with this connection.
+  uint64 delay_time_period = 6 [(gogoproto.moretags) = "yaml:\"delay_time_period\""];
+  // delay height period associated with this connection.
+  uint64 delay_height_period = 7 [(gogoproto.moretags) = "yaml:\"delay_height_period\""];
```

Comment:
This is more tricky. We cannot rely on [protobuf versioning](https://developers.google.com/protocol-buffers/docs/overview#updating) since we construct the connection and then serialize it. We should add the delay height in a backwards compatible way (I think delay height just needs to be changed to optional?)

and 

when we go to serialize our connection in the keeper, we should check its connection version. IF the connection version == 1, then we should use the legacy connection protobuf definition, otherwise we should use this connection definition

The legacy connection should be a copy of the old definition but renamed to `LegacyConnection`. It should only be used to serialize connections. 

We should deserialize using the updated connection. From my understanding, protobuf should still successfully deserialize the old version into the new version 

I'm not 100% sure this will work, but we should find out

cc @cwgoes @fedekunze

Reply:
We should also determine an end date for support of the legacy connection (this is not something we should indefinitely support imo, chains should update especially since there are significant security increases in the next coordinated upgrade)